rfcs/20190829-tfx-container-component-execution.md

```diff
@@ -0,0 +1,555 @@
+# TFX Container Component Execution Proposal
+
+Status         | Proposed
+:------------- | :-------
+**Author(s)**  | Ajay Gopinathan (<a href="mailto:ajaygopinathan@google.com">ajaygopinathan@google.com</a>), Hongye Sun (<a href="mailto:hongyes@google.com">hongyes@google.com</a>), Makoto Uchida (<a href="mailto:muchida@google.com">muchida@google.com</a>), Ruoyu Liu (<a href="mailto:ruoyu@google.com">ruoyu@google.com</a>)
+**Sponsor(s)** | Konstantinos Katsiapis (katsiapis@google.com), Pavel Dournov (dournov@google.com), Ramesh Chandra (rameshuc@google.com)
+**Updated**    | 2019-08-29
+
+## Objective
```

Comment:
Operations within the container may be complex and could benefit from the ml_metadata module.  Can there be a way for the various steps in a container to "log" their events similar to the way components do that now?....In that there wouldn't just be one "component" in the container, but may be multiple operations; each of which would be recorded in the same ml_metadata that the rest of the pipeline uses.